src/schema_manager.h

```diff
@@ -36,6 +36,45 @@ namespace valkey_search {
 constexpr absl::string_view kSchemaManagerMetadataTypeName{"vs_index_schema"};
 constexpr uint32_t kMetadataEncodingVersion = 1;
 
+/*
+
+The SchemaManager uses the MetadataManager to manage a single name space of
+indexes, i.e., map<string, GlobalMetadataEntry>.
```

Comment:
Appreciate the thought going into this. The interoperability is a headache

One question: when someone creates an index in db_num > 0 on Valkey 9, how would a shard running on Valkey 8 behave? I am guessing it would crash, right? I guess that behavior is fine, but we should document this

Another option I guess is that we have namespaces in the metadata manager. Right now we use "vs_index_schema". We could move to "vs_index_schema2"? Nodes on the old versions would just not get the index definitions for those created with vs_index_schema2, which should be fine, and we can register for callbacks of "vs_index_schema" in the new node for backwards compat. Probably a bit more work, but it would prevent the Valkey 8.0 nodes from crashing in such cases